cli/azd/cmd/contracts/env_refresh.go

```diff
@@ -6,8 +6,13 @@ import (
 	"github.com/azure/azure-dev/cli/azd/pkg/infra/provisioning"
 )
 
-// LoginResult is the contract for the output of `azd env refresh`.
+const (
+	EnvRefreshResultType = "envRefreshResult"
+)
+
+// EnvRefreshResult is the contract for the output of `azd env refresh`.
 type EnvRefreshResult struct {
+	Type      string                               `json:"type"`
```

Comment:
I considered introducing an `Envelope` type here which has the `Type` property (and perhaps in that case we'd also add a `Timestamp` property) and a `Data` property, which is the actual event data.

e.g. instead of:

```
{
  type: "envRefreshResult",
  outputs: { ... },
  resources: [ ... ]
}
```

We'd have:  

```
{
  type: "envRefreshResult",
  data: {
    outputs: { ... },
    resources: [ ... ]
  }
}
```

This felt a like it would give us a little more long-term flexibility (e.g. we wouldn't have to worry about what we would do if the object we want to return as a result also have a `type` property, and we could add further metadata without worrying about key conflicts) but it would make the output a little "worse" for interacting with, since in the case of a command like `azd env show`, which would only emit one JSON object right now, you would need to access the `data` property to get the result.

But maybe that slight pain is worthwhile for an easier versioning story on our end.

@philliphoff Do you have any thoughts on this?  As I wrote this comment, I think I am leaning more towards doing the wrapping here, wanted to get a gut check from you as a consumer if that would be totally unreasonable.

Reply:
@ellismg My personal preference is to have the envelope.  From VS Code's perspective, it doesn't technically matter, as both styles can easily be accommodated in TS in a strongly-typed manner using the `type` as a key.  From VS's perspective, the envelope would also likely be cleaner; it's generally easier to define the serialization types such that one property of an object is "dynamic" in whole rather than one object having differing sets of properties based on one property of that same object.  (In that case, a "simpler" approach is to simply have *all* possible properties defined on the object but considered optional, and then the consuming logic apply the "semantics" of matching applicable properties to a given type.  But that approach is harder to understand and maintain over time.)

As you mention, the envelope allows the data to have overlapping-but-semantically-different properties, which would add unnecessary complexity if all possible properties were lumped into a single object.